src/RBDyn/Jacobian.cpp

```diff
@@ -36,6 +40,63 @@ Jacobian::Jacobian(const MultiBody & mb, const std::string & bodyName, const Eig
   jacDot_.resize(6, dof);
 }
 
+Jacobian::Jacobian(const MultiBody & mb,
+                   const std::string & bodyName,
+                   const std::string & refBodyName,
+                   const Eigen::Vector3d & point)
+: jointsPath_(), point_(point), jac_(), jacDot_()
+{
+  body_index_ = mb.sBodyIndexByName(bodyName);
+  ref_index_ = mb.sBodyIndexByName(refBodyName);
+
+  int index = body_index_;
+
+  int dof = 0;
+  bool refindexFound = false;
+  while(!refindexFound && index != -1)
+  {
+    jointsPath_.insert(jointsPath_.begin(), index);
+    dof += mb.joint(index).dof();
+    reverseJoints_.insert(reverseJoints_.begin(), false);
+
+    if(index == ref_index_)
+      refindexFound = true;
+    else
+      index = mb.parent(index);
+  }
+
+  // The two bodies don't belong to the same branch -> start from ref body to the root of the tree
+  if(index != ref_index_)
+  {
+    index = ref_index_;
+    int count = 0;
+    // Add joints until reaching the common node
+    do
+    {
+      jointsPath_.insert(jointsPath_.begin() + count, index);
+      dof += mb.joint(index).dof();
+      reverseJoints_.insert(reverseJoints_.begin(), true);
+
+      index = mb.parent(index);
+      count++;
+    } while(std::find(jointsPath_.begin(), jointsPath_.end(), index) == jointsPath_.end());
+
+    // Delete common joints previously added
+    bool commonJoints = true;
+    while(commonJoints)
+    {
+      const auto joint_index = jointsPath_[static_cast<size_t>(count)];
+      if(joint_index == index) commonJoints = false;
+      dof -= mb.joint(joint_index).dof();
+      jointsPath_.erase(jointsPath_.begin() + count);
+      reverseJoints_.erase(reverseJoints_.begin() + count);
+    }
```

Comment:
I think this can be rewritten in a slightly more compact way:

```cpp
int commonIdx = count; 
while(jointsPaths_[++commonIdx] != index); // Get to the common node
dof -= std::accumulate(jointsPath_.begin() + count, jointsPath_.begin() + commonIdx, 0, [&](int dofC, int idx) { return dofC + mb.joint(idx).dof(); });
jointsPath_.erase(jointsPath_.begin() + count, jointsPath_.begin() + commonIdx);
reverseJoints_erase(jointsPath_.begin() + count, jointsPath_.begin() + commonIdx);
```